stan/math/prim/fun/lb_constrain.hpp

```diff
@@ -22,17 +26,21 @@ namespace math {
  *
  * <p>where \f$L\f$ is the constant lower bound.
  *
- * @tparam T type of Matrix
- * @tparam L type of lower bound
- * @param[in] x Unconstrained Matrix input
+ * @tparam T Scalar.
+ * @tparam L Scalar.
+ * @param[in] x Unconstrained input
  * @param[in] lb Lower bound
  * @return Constrained matrix
  */
-template <typename T, typename L>
+template <typename T, typename L, require_all_stan_scalar_t<T, L>* = nullptr,
+          require_all_not_st_var<T, L>* = nullptr>
 inline auto lb_constrain(const T& x, const L& lb) {
-  auto& lb_ref = to_ref(lb);
-  check_finite("lb_constrain", "lb", value_of(lb_ref));
-  return eval(add(exp(x), lb_ref));
+  if (unlikely(lb == NEGATIVE_INFTY)) {
+    return identity_constrain(x, lb);
```

Comment:
Just return `x` here. Let's remove identity_constrain.

Reply:
We need the identity constrain to return back the correct type. So like if x or lb differ in type `x` get's promoted here

Reply:
Oooof, I see.